addons/base_import_module/models/ir_module.py

```diff
@@ -258,6 +268,125 @@ def module_uninstall(self):
             modules_to_delete.unlink()
         return res
 
+    @api.model
+    def web_search_read(self, domain, specification, offset=0, limit=None, order=None, count_limit=None):
+        res = super().web_search_read(domain, specification, offset=offset, limit=limit, order=order, count_limit=count_limit)
+        if any(dom for dom in domain if dom[0] == 'module_type' and dom[-1] != 'official'):
+            module_type = [dom for dom in domain if dom[0] == 'module_type'][0][-1]
+            fields = list(specification.keys())
+            modules_list = self._get_modules_from_apps(fields, module_type, False)
+            res['length'] += len(modules_list)
+            res['records'].extend(modules_list)
+        return res
+
+    def more_info(self):
+        return {
+            'name': _('Apps'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'ir.module.module',
+            'view_mode': 'form',
+            'res_id': self.id,
+            'context': self.env.context,
+        }
+
+    def web_read(self, specification):
+        fields = list(specification.keys())
+        module_type = self.env.context.get('module_type', 'official')
+        if module_type != 'official':
+            modules_list = self._get_modules_from_apps(fields, module_type, self.env.context.get('module_name'))
+            return modules_list
+        else:
+            return super().web_read(specification)
+
+    @api.model
+    def _get_modules_from_apps(self, fields, module_type, module_name):
+        payload = {
+            'series': major_version,
+            'module_fields': fields,
+            'module_type':module_type,
+            'module_name': module_name
+        }
+        headers = {'Content-Type': 'application/json'}
+        resp = requests.post(
+            f"{APPS_URL}/loempia/listdatamodules",
+            data=dumps({'params': payload}),
+            headers=headers,
```

Comment:
```suggestion
            headers=headers, timeout=5,
```
(or whatever time it resonably takes)